bcda/servicemux.go

```diff
@@ -86,7 +89,39 @@ func (sm *ServiceMux) AddServer(s *http.Server, m string) {
 }
 
 func (sm *ServiceMux) Serve() {
-	m := cmux.New(tcpKeepAliveListener{sm.Listener.(*net.TCPListener)})
+	tlsCertPath := os.Getenv("BCDA_TLS_CERT")
+	tlsKeyPath := os.Getenv("BCDA_TLS_KEY")
+
+	if tlsCertPath != "" && tlsKeyPath != "" {
+		sm.serveHTTPS(tlsCertPath, tlsKeyPath)
+	} else if tlsCertPath != "" {
+		panic("TLS certificate path provided, but no key")
+	} else if tlsKeyPath != "" {
+		panic("TLS key path provided, but no certificate")
+	} else {
+		log.Warn("TLS not enabled")
+		sm.serveHTTP()
+	}
+}
+
+func (sm *ServiceMux) serveHTTPS(tlsCertPath, tlsKeyPath string) {
+	certificate, err := tls.LoadX509KeyPair(tlsCertPath, tlsKeyPath)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	config := &tls.Config{
```

Comment:
Side note: with this configuration, when `https` is enabled, we _only_ support `https`. Should we also accept `http` requests and simply redirect them to their `https` equivalent? My instincts are telling me yes, it would be smart to add this in. This is another question that comes straight from link in my previous comment.